core/kernel/dt.c

```diff
@@ -139,6 +142,24 @@ int dt_map_dev(const void *fdt, int offs, vaddr_t *base, size_t *size,
 	return 0;
 }
 
+TEE_Result dt_unmap_dev(const void *fdt, int offs, vaddr_t base,
+			enum dt_map_dev_directive mapping)
+{
+	size_t sz = 0;
+	int st = DT_STATUS_DISABLED;
+
+	st = _fdt_get_status(fdt, offs);
+	if (st == DT_STATUS_DISABLED)
+		return TEE_ERROR_BAD_STATE;
+
+	sz = _fdt_reg_size(fdt, offs);
+	if (sz == DT_INFO_INVALID_REG_SIZE)
+		return -1;
+
+	return core_mmu_remove_mapping(dt_to_type(st, mapping), (void *)base,
+				       sz);
```

Comment:
i don't think this function is always reliable: `core_mmu_add_mapping()` does not count consumers of a mapped area and reuses already mapped VA ranges when possible. `core_mmu_remove_mapping()` may therefore unmap a VA range used by another component.

Reply:
Hum ok, so what should I do for that series ? Remove this and assume even if the driver fails we keep the memory mapping ?